site/docs/blog/posts/2024-05-new-cli.md

```diff
@@ -0,0 +1,119 @@
+---
+date: 2024-05-02
+authors:
+  - snazy
+---
+
+# New Nessie CLI tool
+
+There is a new CLI tool for Nessie, replacing the old Python based CLI.
+<!-- more -->
+
+![Nessie CLI](../../img/cli-intro.png)
+
+`pynessie` was _the_ command line tools when working against Nessie from
+a command line. It exists since the early days of Nessie.  Since then,
+`pynessie` did not change very much and is still using the Nessie REST
+API v1, which is deprecated and will be removed at some point.
+`pynessie` also does not have support for the more complex OAuth2
+device-code and authorization-code flows. 
+
+Updating `pynessie` to work with Nessie REST API v2 basically requires
+to re-implement a lot of code, starting with JSON de/serialization,
+model objects, etc.
+
+We were thinking about how to deal with a CLI tool for Nessie,
+considering a bunch of options and choices:
+
+## Shall we continue with Python or use Java?
+
+You may say: Python, that's it! Much easier to run! With _easier_
+people usually think of "just running it" from bash or zsh. But is
+it really easier run? You have to setup a virtual-env and have the
```

Comment:
```suggestion
it really easier to run? You have to setup a virtual-env and have the
```

Reply:
Not sure if this argument is a convincing one: virtual-env is not strictly speaking necessary, and besides, python devs all know virtual-env by heart.